### ISSUE TITLE
Installing Inkscape resulted in broken system

### ISSUE THREAD

**Issue by dviererbe** [ubuntu/app-center#1597]
This weekend I tried to install the deb version of Inkscape through the app-center.

Here is the apt log of what exactly happened:
```
Start-Date: 2024-02-03  13:59:53
Commandline: packagekit role='install-packages'
Requested-By: dviererbe (1000)
Install: libcups2:i386 (2.4.6-0ubuntu3, automatic), libgomp1:i386 (13.2.0-4ubuntu3, automatic), libwebpmux3:i386 (1.2.4-0.3, automatic), libltdl7:i386 (2.4.7-7, automatic), libsoup2.4-1:i386 (2.74.3-1, automatic), libatkmm-1.6-1v5:i386 (2.28.3-1, automatic), libdatrie1:i386 (0.2.13-2, automatic), libthai0:i386 (0.1.29-1build1, automatic), libpotrace0:i386 (1.16-2, automatic), libxrender1:i386 (1:0.9.10-1.1, automatic), libproxy1v5:i386 (0.4.18-1.2, automatic), libnettle8:i386 (3.9.1-2, automatic), libpango-1.0-0:i386 (1.51.0+ds-2, automatic), libtasn1-6:i386 (4.19.0-3, automatic), libwayland-cursor0:i386 (1.22.0-2.1, automatic), libselinux1:i386 (3.5-1, automatic), libaom3:i386 (3.6.1-1, automatic), libjpeg-turbo8:i386 (2.1.5-2ubuntu1, automatic), libavahi-common-data:i386 (0.8-10ubuntu1.1, automatic), libheif1:i386 (1.16.2-2ubuntu1, automatic), libcolord2:i386 (1.4.6-3, automatic), libx265-199:i386 (3.5-2, automatic), libcairo2:i386 (1.18.0-1, automatic), python3-importlib-resources:amd64 (6.0.1-1, automatic), libxcursor1:i386 (1:1.2.1-1, automatic), libheif-plugin-libde265:i386 (1.16.2-2ubuntu1, automatic), libdb5.3:i386 (5.3.28+dfsg2-2, automatic), libp11-kit0:i386 (0.25.0-4ubuntu1, automatic), libdeflate0:i386 (1.18-1, automatic), libatk-bridge2.0-0:i386 (2.50.0-1, automatic), libjpeg8:i386 (8c-2ubuntu11, automatic), libfftw3-double3:i386 (3.3.10-1ubuntu1, automatic), libcairomm-1.0-1v5:i386 (1.14.4-2, automatic), libdav1d6:i386 (1.2.1-2, automatic), libfribidi0:i386 (1.0.13-3, automatic), python3-typing-extensions:amd64 (4.7.1-1, automatic), libgpm2:i386 (1.20.7-10build1, automatic), libwebpdemux2:i386 (1.2.4-0.3, automatic), libtiff6:i386 (4.5.1+git230720-1ubuntu1, automatic), libsigsegv2:amd64 (2.14-1ubuntu1, automatic), libpixman-1-0:i386 (0.42.2-1, automatic), libsystemd0:i386 (253.5-1ubuntu6.1, automatic), libfreetype6:i386 (2.13.1+dfsg-1, automatic), libmount1:i386 (2.39.1-4ubuntu2, automatic), librsvg2-common:i386 (2.54.7+dfsg-2, automatic), libxcomposite1:i386 (1:0.4.5-1build2, automatic), libnuma1:i386 (2.0.16-1, automatic), libgspell-1-2:i386 (1.12.2-1, automatic), libwpd-0.10-10:i386 (0.10.3-2build1, automatic), libdbus-1-3:i386 (1.14.10-1ubuntu1, automatic), libxdamage1:i386 (1:1.1.6-1, automatic), libpython3.11-minimal:i386 (3.11.6-3, automatic), libgdk-pixbuf-2.0-0:i386 (2.42.10+dfsg-1build1, automatic), libboost-filesystem1.74.0:i386 (1.74.0+ds1-22ubuntu1, automatic), libcairo-gobject2:i386 (1.18.0-1, automatic), libgraphite2-3:i386 (1.3.14-1build2, automatic), libfontconfig1:i386 (2.14.2-4ubuntu1, automatic), libhunspell-1.7-0:i386 (1.7.2+really1.7.2-10, automatic), libdouble-conversion3:i386 (3.3.0-1, automatic), libmagick++-6.q16-8:i386 (8:6.9.11.60+dfsg-1.6ubuntu1, automatic), libudev1:i386 (253.5-1ubuntu6.1, automatic), libgpg-error0:i386 (1.47-2, automatic), libcdr-0.1-1:i386 (0.1.7-1, automatic), libpng16-16:i386 (1.6.40-1, automatic), libreadline8:i386 (8.2-1.3, automatic), libheif-plugin-x265:i386 (1.16.2-2ubuntu1, automatic), libheif-plugin-aomenc:i386 (1.16.2-2ubuntu1, automatic), libxrandr2:i386 (2:1.5.2-2, automatic), libpangomm-1.4-1v5:i386 (2.46.3-1, automatic), gawk:amd64 (1:5.2.1-2, automatic), libcap2:i386 (1:2.66-4ubuntu1, automatic), libaspell15:i386 (0.60.8-5, automatic), libpangocairo-1.0-0:i386 (1.51.0+ds-2, automatic), libxkbcommon0:i386 (1.5.0-1, automatic), libatk1.0-0:i386 (2.50.0-1, automatic), libpoppler-glib8:i386 (23.08.0-2ubuntu1, automatic), libxcb-render0:i386 (1.15-1, automatic), libuuid1:i386 (2.39.1-4ubuntu2, automatic), libgcrypt20:i386 (1.10.2-3ubuntu1, automatic), policykit-1-gnome:amd64 (0.105-7ubuntu3, automatic), libatspi2.0-0:i386 (2.50.0-1, automatic), libmagickwand-6.q16-6:i386 (8:6.9.11.60+dfsg-1.6ubuntu1, automatic), libpoppler130:i386 (23.08.0-2ubuntu1, automatic), libnss3:i386 (2:3.92-1, automatic), libpangoft2-1.0-0:i386 (1.51.0+ds-2, automatic), libpython3.11-stdlib:i386 (3.11.6-3, automatic), enchant-2:i386 (2.3.3-2, automatic), libsqlite3-0:i386 (3.42.0-1ubuntu0.1, automatic), libwebp7:i386 (1.2.4-0.3, automatic), libxslt1.1:i386 (1.1.35-1, automatic), libgmp10:i386 (2:6.3.0+dfsg-2ubuntu4, automatic), libglibmm-2.4-1v5:i386 (2.66.6-2, automatic), libpcre2-8-0:i386 (10.42-4, automatic), libcrypt1:i386 (1:4.4.36-2, automatic), libavahi-common3:i386 (0.8-10ubuntu1.1, automatic), libwmf-bin:amd64 (0.2.13-1, automatic), librsvg2-2:i386 (2.54.7+dfsg-2, automatic), libmagickcore-6.q16-6:i386 (8:6.9.11.60+dfsg-1.6ubuntu1, automatic), librevenge-0.0-0:i386 (0.0.5-3, automatic), libxinerama1:i386 (2:1.1.4-3, automatic), libjbig0:i386 (2.1-6.1ubuntu1, automatic), libopenjp2-7:i386 (2.5.0-2, automatic), libwayland-egl1:i386 (1.22.0-2.1, automatic), libpython3-stdlib:i386 (3.11.4-5, automatic), libglib2.0-0:i386 (2.78.0-2, automatic), libgsl27:i386 (2.7.1+dfsg-6ubuntu1, automatic), libbrotli1:i386 (1.0.9-2build8, automatic), libgtk-3-0:i386 (3.24.38-5ubuntu1, automatic), libgtkmm-3.0-1v5:i386 (3.24.8-2, automatic), fig2dev:amd64 (1:3.2.8b-3, automatic), libvisio-0.1-1:i386 (0.1.7-1build7, automatic), libheif-plugin-dav1d:i386 (1.16.2-2ubuntu1, automatic), libhogweed6:i386 (3.9.1-2, automatic), libharfbuzz0b:i386 (8.0.1-1, automatic), libblkid1:i386 (2.39.1-4ubuntu2, automatic), libgnutls30:i386 (3.8.1-4ubuntu1.2, automatic), libenchant-2-2:i386 (2.3.3-2, automatic), liblz4-1:i386 (1.9.4-1, automatic), libgc1:i386 (1:8.2.4-1, automatic), libncursesw6:i386 (6.4+20230625-2, automatic), libbz2-1.0:i386 (1.0.8-5build1, automatic), libgslcblas0:i386 (2.7.1+dfsg-6ubuntu1, automatic), libwayland-client0:i386 (1.22.0-2.1, automatic), liblcms2-2:i386 (2.14-2, automatic), libwpg-0.3-3:i386 (0.3.4-3, automatic), libde265-0:i386 (1.0.12-2, automatic), libsigc++-2.0-0v5:i386 (2.12.0-1, automatic), libepoxy0:i386 (1.5.10-1, automatic), liblqr-1-0:i386 (0.4.2-2.1, automatic), libavahi-client3:i386 (0.8-10ubuntu1.1, automatic), glib-networking:i386 (2.78.0-1, automatic), inkscape:i386 (1.2.2-2ubuntu4), libpsl5:i386 (0.21.2-1, automatic), libxi6:i386 (2:1.8-1build1, automatic), libnspr4:i386 (2:4.35-1.1, automatic)
Remove: reportbug:amd64 (11.6.0ubuntu2), dput:amd64 (1.1.3ubuntu3), network-manager-openvpn-gnome:amd64 (1.10.2-3), orca:amd64 (45.0-1ubuntu1), networkd-dispatcher:amd64 (2.2.4-1), python3-gi:amd64 (3.46.0-1), texlive-pictures:amd64 (2023.20230613-3), network-manager-openvpn:amd64 (1.10.2-3), nautilus-extension-gnome-terminal:amd64 (3.49.92-2ubuntu1), python3-fido2:amd64 (0.9.1-1), python3-distupgrade:amd64 (1:23.10.14), python3-brlapi:amd64 (6.5-7ubuntu2), python3-ykman:amd64 (4.0.9-1), update-manager:amd64 (1:23.10.1), usb-creator-common:amd64 (0.3.17), python3-sphinx:amd64 (5.3.0-7), python3-cffi-backend:amd64 (1.15.1-5build1), network-manager-config-connectivity-ubuntu:amd64 (1.44.2-1ubuntu1.2), apport-gtk:amd64 (2.27.0-0ubuntu5), ibus-table:amd64 (1.17.2-1), python3-lazr.restfulclient:amd64 (0.14.5-1), python3-ibus-1.0:amd64 (1.5.29~rc1-1), deja-dup:amd64 (44.2-2), python3-netplan:amd64 (0.107-5ubuntu0.2), ubuntu-drivers-common:amd64 (1:0.9.7.6), ubuntu-release-upgrader-gtk:amd64 (1:23.10.14), python3-reportbug:amd64 (11.6.0ubuntu2), usb-creator-gtk:amd64 (0.3.17), python3-debianbts:amd64 (4.0.1), os-prober:amd64 (1.81ubuntu3), sssd-proxy:amd64 (2.9.1-2ubuntu2), ubuntu-advantage-desktop-daemon:amd64 (1.10), ibus-table-cangjie3:amd64 (1.8.12-1), ibus-table-cangjie5:amd64 (1.8.12-1), gnome-shell-pomodoro:amd64 (0.24.0-1~exp1), language-selector-gnome:amd64 (0.224), software-properties-common:amd64 (0.99.39), rhythmbox-plugins:amd64 (3.4.7-1ubuntu1), aptdaemon:amd64 (1.1.1+bzr982-0ubuntu44), sssd-ad-common:amd64 (2.9.1-2ubuntu2), system-config-printer-udev:amd64 (1.5.18-1ubuntu3), python3.11-venv:amd64 (3.11.6-3), sssd-ipa:amd64 (2.9.1-2ubuntu2), python3-aptdaemon:amd64 (1.1.1+bzr982-0ubuntu44), xserver-xorg:amd64 (1:7.7+23ubuntu2), python3-software-properties:amd64 (0.99.39), network-manager-gnome:amd64 (1.32.0-3ubuntu1), gnome-shell:amd64 (45.2-0ubuntu1), python3-notify2:amd64 (0.3-5), python3-pycurl:amd64 (7.45.2-4), python3-gdbm:amd64 (3.11.5-1), language-selector-common:amd64 (0.224), system-config-printer-common:amd64 (1.5.18-1ubuntu3), ubuntu-desktop:amd64 (1.524.1), gdm3:amd64 (45~beta-1ubuntu1), network-manager-pptp-gnome:amd64 (1.2.12-2), python3-jsonschema:amd64 (4.10.3-2ubuntu1), totem-plugins:amd64 (43.0-2ubuntu1), gnome-shell-extension-desktop-icons-ng:amd64 (46+really47.0.7-0ubuntu1), python3-oauthlib:amd64 (3.2.2-1), ubuntu-pro-client-l10n:amd64 (30~23.10), python3-pyatspi:amd64 (2.46.0-2), gnome-shell-extension-appindicator:amd64 (57-1), cloud-init:amd64 (23.3.3-0ubuntu0~23.10.1), update-notifier:amd64 (3.192.66), python3-mako:amd64 (1.2.4+ds-2), python3-cairo:amd64 (1.24.0-3), duplicity:amd64 (1.2.2-1ubuntu1), sssd-krb5-common:amd64 (2.9.1-2ubuntu2), autopkgtest:amd64 (5.28ubuntu1), update-manager-core:amd64 (1:23.10.1), network-manager:amd64 (1.44.2-1ubuntu1.2), python3-jinja2:amd64 (3.1.2-1ubuntu0.23.10.1), python3-apt:amd64 (2.6.0ubuntu1), software-properties-gtk:amd64 (0.99.39), enchant-2:amd64 (2.3.3-2), python3-dev:amd64 (3.11.4-5), python3-netifaces:amd64 (0.11.0-2build1), ibus-table-cangjie-big:amd64 (1.8.12-1), python3-gpg:amd64 (1.18.0-3ubuntu2), xorg:amd64 (1:7.7+23ubuntu2), python3-markupsafe:amd64 (2.1.3-1), screen-resolution-extra:amd64 (0.18.3), nvidia-settings:amd64 (510.47.03-0ubuntu1), ubuntu-session:amd64 (45.0-1ubuntu1), network-manager-pptp:amd64 (1.2.12-2), python3-pil:amd64 (10.0.0-1ubuntu0.1), python3-pysimplesoap:amd64 (1.16.2-5), hplip:amd64 (3.22.10+dfsg0-2ubuntu1), python3-sss:amd64 (2.9.1-2ubuntu2), yubikey-manager-qt:amd64 (1.2.4-1), python3-uno:amd64 (4:7.6.4-0ubuntu0.23.10.1), python3-nacl:amd64 (1.5.0-4), python3-yaml:amd64 (6.0.1-1), rhythmbox-plugin-alternative-toolbar:amd64 (0.20.3-1), ubuntu-dev-tools:amd64 (0.196), gnome-shell-extension-ubuntu-tiling-assistant:amd64 (44-1ubuntu1), gnome-shell-extension-ubuntu-dock:amd64 (87ubuntu2), python3.11-dev:amd64 (3.11.6-3), sssd:amd64 (2.9.1-2ubuntu2), gnome-control-center:amd64 (1:45.0-1ubuntu3.1), ubuntu-desktop-minimal:amd64 (1.524.1), python3-update-manager:amd64 (1:23.10.1), netplan.io:amd64 (0.107-5ubuntu0.2), python3-apport:amd64 (2.27.0-0ubuntu5), python3-keyring:amd64 (24.2.0-1), python3-commandnotfound:amd64 (23.04.0), texlive-xetex:amd64 (2023.20230613-3), pre-commit:amd64 (3.3.3-2), python3-aptdaemon.gtk3widgets:amd64 (1.1.1+bzr982-0ubuntu44), python3-cups:amd64 (2.0.1-5build3), memtest86+:amd64 (6.20-3), gnome-terminal:amd64 (3.49.92-2ubuntu1), sssd-krb5:amd64 (2.9.1-2ubuntu2), ibus:amd64 (1.5.29~rc1-1), python3-bcrypt:amd64 (3.2.2-1), unattended-upgrades:amd64 (2.9.1+nmu3ubuntu2), ubuntu-release-upgrader-core:amd64 (1:23.10.14), command-not-found:amd64 (23.04.0), python3-cryptography:amd64 (38.0.4-4ubuntu0.23.10.1), ubuntu-advantage-tools:amd64 (30~23.10), python3-pyscard:amd64 (2.0.7-2), sssd-ad:amd64 (2.9.1-2ubuntu2), python3-launchpadlib:amd64 (1.11.0-1), python3-pyrsistent:amd64 (0.18.1-1build3), python3-dbus:amd64 (1.3.2-5), sssd-common:amd64 (2.9.1-2ubuntu2), texlive-latex-extra:amd64 (2023.20230613-2), python3-ubuntutools:amd64 (0.196), sssd-ldap:amd64 (2.9.1-2ubuntu2), python3-cupshelpers:amd64 (1.5.18-1ubuntu3), apport:amd64 (2.27.0-0ubuntu5), python3-paramiko:
```

It looks to me that the `i386` version of Inkscape (instead of `amd64`) was selected to install and to fulfill the dependencies apt had to remove a bunch of essential packages.

If I would have done this through the terminal apt would ask me if I really want to remove all these packages, but the apt-center doesn't and left me with a broken system. 

I am aware that the dependency resolution is the responsibility of apt, but why
1. was the `i386` version selected by app-center?
2. is there no confirmation that I really want to go ahead with uninstalling essential packages?

**Comment by AsciiWolf** [comment]
/cc @Feichtmeier @d-loose

Is this still happening? If so, it should be fixed ASAP.

**Comment by Feichtmeier** [comment]
I don't lead this project here anymore but I'll forward it.
If it's really urgent maybe also consider to create a launchpad issue

**Comment by d-loose** [comment]
@dviererbe which version of app-center are you running (`snap info snap-store`)?

This got [fixed](https://github.com/ubuntu/app-center/pull/1486) back in November. It looks like the `latest/stable/ubuntu-23.10` channel still points to an old version from the end of September, though. @tim-hm did we not promote that fix to 23.10 while we were in Riga?

**Comment by dviererbe** [comment]
```
name:    snap-store
summary: Snap Store is a graphical desktop application for discovering,
  installing and managing snaps on Linux.
publisher: Canonical**
contact:   https://bugs.launchpad.net/snap-store/
license:   unset
description: |
  Snap Store showcases featured and popular applications with useful
  descriptions, ratings, reviews and screenshots.
  
  
  Applications can be found either through browsing categories
  or by searching.
  
  
  Snap Store can also be used to switch channels, view and alter snap
  permissions and view and submit reviews and ratings.
  
  
  Snap Store is based on GNOME Software, optimized for the Snap experience.
commands:
  - snap-store
snap-id:      gjf3IPXoRiipCu9K0kVu52f0H56fIksg
tracking:     latest/stable/ubuntu-23.10
refresh-date: 2023-11-27
installed:    0+git.e118b05 (1046) 11MB -
```
> This got [fixed](https://github.com/ubuntu/app-center/pull/1486) back in November.

@d-loose It looks like the snap store snap did not refresh since November. My memory could be wrong, but I do not remember seeing any snap-store updates that I could install.

NOTE: I reinstalled Ubuntu since then onto another drive. I can still boot into the broken Ubuntu installation or look into log files. Fell free to ask if I should provide additional information.

**Comment by dviererbe** [comment]
When I run `snap info snap-store` on my fresh install I see:
```
name:      snap-store
summary:   App Center
publisher: Canonical✓
store-url: https://snapcraft.io/snap-store
license:   unset
description: |
  App Center
commands:
  - snap-store
snap-id:      gjf3IPXoRiipCu9K0kVu52f0H56fIksg
tracking:     latest/stable/ubuntu-23.10
refresh-date: 2023-10-16
channels:
  latest/stable:     41.3-71-g709398e 2023-04-28  (959) 12MB -
  latest/candidate:  41.3-76-g2e8f3b0 2023-10-12 (1058) 12MB -
  latest/beta:       ↑                                       
  latest/edge:       0+git.eefeb82    2024-01-17 (1099) 10MB -
  preview/stable:    –                                       
  preview/candidate: 0.2.7-alpha      2023-02-02  (864) 10MB -
  preview/beta:      ↑                                       
  preview/edge:      0.3.0-alpha      2023-08-14 (1017) 11MB -
installed:           0+git.e118b05               (1046) 11MB -
```

Note `refresh-date: 2023-10-16`. Sorry if I misread the output, but doesn't this mean that my fresh install has an even older version?

**Comment by tim-hm** [comment]
I thought we had ... but clearly not the case (I tripled checked locally). We'll need someone with permission to promote 1099 to `latest/stable/ubuntu-23.10`.

**Comment by d-loose** [comment]
Fixed in both `latest/stable/ubuntu-23.10` and `latest/stable/ubuntu-24.04`.
